src/jit/hwintrinsicxarch.cpp

```diff
@@ -219,19 +219,67 @@ unsigned Compiler::simdSizeOfHWIntrinsic(NamedIntrinsic intrinsic, CORINFO_SIG_I
 }
 
 //------------------------------------------------------------------------
-// numArgsOfHWIntrinsic: get the number of arguments
+// numArgsOfHWIntrinsic: get the number of arguments based on table and
+// if numArgs is -1 check number of arguments using GenTreeHWIntrinsic
+// node unless it is nullptr
 //
 // Arguments:
-//    intrinsic -- id of the intrinsic function.
+//    intrinsic -- id of the intrinsic function
+//    node      -- GenTreeHWIntrinsic* node with nullptr default value
 //
 // Return Value:
 //     number of arguments
 //
-int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic)
+int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, GenTreeHWIntrinsic* node)
 {
     assert(intrinsic != NI_Illegal);
     assert(intrinsic > NI_HW_INTRINSIC_START && intrinsic < NI_HW_INTRINSIC_END);
-    return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
+
+    int numArgs = hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
+    // TODO_XARCH-BUG: several intrinsics having -1 numArgs do not handle correctly
+    // real number of arguments - limiting slow path only to Sse2.Shuffle
+    // https://github.com/dotnet/coreclr/issues/16714
```

Comment:
I am not comfortable with checking in this workaround. I believe that @fiigii suggested the right fix. I think these asserts:
```
            assert(numArgs != 0);
            assert(numArgs != 1);
```
Should be replaced with `assert(numArgs >= 2);` below the `if (info->srcCount >= 2)` check.